File: src/clients/Java/src/main/java/br/com/takenet/takeio/entities/Collection.java
package br.com.takenet.takeio.entities;

public class Collection<T> {		
	private ResultType<T> result;
	
	public Collection() {
	}

	public ResultType<T> getResult() {
		return result;
	}

	public void setResult(ResultType<T> result) {
		this.result = result;
	}
	
}

File: src/clients/CSharp/TakeIoLib/Entities/Collection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TakeIoLib.Entities
{
    public class Collection<T>
    {
        public class ResultType<Inner>
        {
            public virtual int? TotalResults { get; set; }
            public virtual int? StartIndex { get; set; }
            public virtual int? ItemsPerPage { get; set; }
            public virtual IList<Inner> Entry { get; set; }
        }

        public virtual ResultType<T> Result { get; set; }
    }
}
